rust/src/editor.rs

```diff
@@ -177,6 +178,29 @@ impl Editor {
         }
     }
 
+    fn cursor_start(&mut self) {
+        let start = self.view.sel_min() - self.col;
+        self.set_cursor(start, true);
+    }
+
+    fn cursor_end(&mut self) {
+        let current = self.view.sel_max();
+        let rope = self.text.clone();
+        let mut cursor = Cursor::new(&rope, current);
+        match cursor.next::<LinesMetric>() {
+            None => { self.set_cursor(current, true); },
+            Some(offset) => {
+                if cursor.is_boundary::<LinesMetric>() {
+                    if let Some(new) = rope.prev_grapheme_offset(offset) {
+                        self.set_cursor(new + 1, true);
```

Comment:
This is strictly for going back before the newline at the end of the line, as it is ctrl-e will advance to the beginning of the next line. I'll push the change to `is_boundary` now.